crates/matrix-synapse/src/lib.rs

```diff
@@ -189,25 +207,57 @@ impl HomeserverConnection for SynapseConnection {
         err(Display),
     )]
     async fn is_localpart_available(&self, localpart: &str) -> Result<bool, Self::Error> {
+        #[derive(Deserialize)]
+        struct UsernameAvailableResponse {
+            available: bool,
+        }
```

Comment:
The other response structures are defined at the top of the file, for consitency could you move that up there?

Reply:
sure. It's just a personal style I like for these single-use structs, to make them function-local, but yeah indeed there are others here so let's do that